core/src/test/java/org/elasticsearch/cluster/routing/PrimaryAllocationIT.java

```diff
@@ -109,7 +109,12 @@ private void createStaleReplicaScenario() throws Exception {
         logger.info("--> check that old primary shard does not get promoted to primary again");
         // kick reroute and wait for all shard states to be fetched
         client(master).admin().cluster().prepareReroute().get();
-        assertBusy(() -> assertThat(internalCluster().getInstance(GatewayAllocator.class, master).getNumberOfInFlightFetch(), equalTo(0)));
```

Comment:
I gave it the obvious fix without any special sugar. and plenty of assertBusy's, actually most of them, are written this way.

again, as it was a compiler failure, the style is unimportant to me really. More important was to get mmap working :)